### PR TITLE
feat!: local-ic remove relayer file dependency

### PR DESCRIPTION
**Pull request by Reecepbcups**
ref #795

Previously it was required for developers to maintain a relayer.json file with the image information. For most use cases this is excessive as the defaults work. Instead it should be a startup option to override default relayer image information.

If none are specific, the latest cosmos relayer image is used.

Only a breaking change if a team uses a custom relayer image for local-ic

### CODE REVIEW

File: local-interchain/python/helpers/file_cache.py [hunk 1]
```diff
@@ -33,6 +34,7 @@ def get_chain_start_time_from_logs() -> int:
     def get_cache_or_default(contracts: dict, ictest_chain_start: int) -> dict:
         cache_time: str | int = 0
         if os.path.exists(contracts_json_path):
+            os.makedirs(os.path.dirname(contracts_json_path), exist_ok=True)
```
Comment (jtieri) [inline local-interchain/python/helpers/file_cache.py:37]: i don't Python so can't speak to these changes but i will assume these two lines are correct

**Review by jtieri — APPROVED** [review]
changes look straightforward enough. 

what's up with the empty `.gitkeep` file? is there some assumption that this empty directory needs to exist for tests to run or something?

File: local-interchain/python/helpers/file_cache.py (hunk 1 above)
Comment (Reecepbcups) [reply local-interchain/python/helpers/file_cache.py:37]: yes, just creates a nested directory if it does not exist. if it does, it does nothing

**Comment by Reecepbcups** [comment]
let's see

**Pull request merged**